tests/components/remote/test_init.py

```diff
@@ -117,3 +117,13 @@ def test_learn_command(self):
         assert call.domain == remote.DOMAIN
         assert call.service == SERVICE_LEARN_COMMAND
         assert call.data[ATTR_ENTITY_ID] == "entity_id_val"
+
+
+def test_deprecated_base_class(caplog):
+    """Test deprecated base class."""
+
+    class CustomRemote(remote.RemoteDevice):
+        pass
+
+    CustomRemote()
+    assert "RemoteDevice is deprecated, modify CustomRemote" in caplog.text
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5418277)